### PR TITLE
Scale mouse delta by zoom when using shift.

### PR DESCRIPTION
**Pull request by ZehMatt**
This is something that bothered me for a while, the current behavior is this:
2018-08-10_01-14-14bc0f9f495f44ff7e1
Which clearly does not consider the zoom.

Thats what I believe it should look like:
2018-08-10_01-13-03190527e718915af29

### CODE REVIEW

**Comment by Broxzier** [comment]
I remember running into an issue with the height overflowing when I tried solving this before. It may be an issue already, only much easier to replicate now.

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 1]
```diff
@@ -1288,6 +1309,7 @@ static void sub_6E1F34(
                         z += gSceneryShiftPressZOffset;
 
                         z = std::max<int16_t>(z, 16);
+                        z = std::min<int16_t>(z, maxPossibleHeight);
```
Comment (Broxzier) [inline src/openrct2-ui/windows/TopToolbar.cpp]: `std::min` and `std::max` can be combined with `Math::Clamp`

Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/TopToolbar.cpp]: We should also be able to use `std::clamp` now that we're on C++17. `Math::Clamp` should probably be phased out entirely. Note that the order of arguments differs.

Comment (Broxzier) [reply src/openrct2-ui/windows/TopToolbar.cpp]: Right, that'd be even nicer! Technically the order doesn't really matter, as long as the last argument is the max, but it'd be nicer to match the values with the argument's names of course.

File: src/openrct2-ui/windows/RideConstruction.cpp [hunk 2]
```diff
@@ -31,6 +31,7 @@
 #include <openrct2/world/Entrance.h>
 #include <openrct2/world/Footpath.h>
 #include <openrct2/world/Park.h>
+#include <limits>
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/RideConstruction.cpp]: Should go before the `openrct2` includes, per clang-format.

**Pull request merged**